case-1/src/main/java/com/company/app/pvmodulelayouts/DefaultPvModuleLayouts.java

```diff
@@ -4,18 +4,23 @@
 import com.company.app.PvModuleLayouts;
 import com.company.app.PvModulePosition;
 import com.company.app.RoofFace;
-import com.company.app.pvmodulelayouts.layoutspecification.PvModulesMountedOnRoofFacePlaneLayoutSpecification;
-import com.company.app.pvmodulelayouts.layoutspecification.PvModulesMountedOnTiltedRacksLayoutSpecification;
 
 import java.util.List;
+import java.util.function.BiFunction;
 import java.util.stream.Stream;
 
 public class DefaultPvModuleLayouts implements PvModuleLayouts {
+    private final BiFunction<RoofFace, PvModuleDefinition, LayoutSpecification> layoutSpecificationConstructor;
+
+    public DefaultPvModuleLayouts(
+            BiFunction<RoofFace, PvModuleDefinition, LayoutSpecification> layoutSpecificationConstructor
+    ) {
+        this.layoutSpecificationConstructor = layoutSpecificationConstructor;
+    }
```

Comment:
If the constructor required more parameters, we'd probably have to introduce something like this

```
interface LayoutSpecificationConstructor {
    LayoutSpecification construct(
        RoofFace roofFace,
        PvModuleDefinition pvModuleDefinition,
        Object someOtherArg1,
        Object someOtherArg2
     );
}
```
probably as a nested interface inside `LayoutSpecification`